vendor/github.com/ethereum/go-ethereum/internal/ethapi/api.go

```diff
@@ -956,6 +956,14 @@ func (s *PublicTransactionPoolAPI) GetRawTransactionByBlockHashAndIndex(ctx cont
 
 // GetTransactionCount returns the number of transactions the given address has sent for the given block number
 func (s *PublicTransactionPoolAPI) GetTransactionCount(ctx context.Context, address common.Address, blockNr rpc.BlockNumber) (*hexutil.Uint64, error) {
+	// go-ethereum issue https://github.com/ethereum/go-ethereum/issues/2880
```

Comment:
We need to add this to go-ethereum patches. cc @divan

Reply:
I think there is a way to get rid of it. we can maintain nonce locally for each address that status is using. increment it on every successful transaction, and reset in case if it was changed remotely. it is easy with new addr locker and will prevent issues both with upstream and local nodes.

will do a separate pr after additional testing

Reply:
It may introduce a lot of complexity into status-go and something that potentially can break in the future if go-ethereum implementation changes. But if you find some elegant solution that would be awesome.

I have seen this PR https://github.com/ethereum/go-ethereum/pull/15794 and read the whole discussion in https://github.com/ethereum/go-ethereum/issues/2880 and it looks like a separate JSON-RPC method is the best solution that can preserve backward compatibility of `eth_getTransactionCount` and allows to get the number of pending transactions from the pool.

Reply:
adding a separate method also seemed like a good idea to me, but it may take some time until it will be merged and a new geth release will be available. and even then status app needs to be always aware that upstream node is not an old geth release or another client, e.g parity.

Reply:
Yeah, that should be implemented via patches to keep track of our changes into the `go-ethereum`. This is blocked by https://github.com/status-im/status-go/pull/492, which will be merged tomorrow (internally agreed deadline).